merlin/dag/selector.py

```diff
@@ -130,14 +142,19 @@ def __rshift__(self, operator):
     def __eq__(self, other):
         if not isinstance(other, ColumnSelector):
             return False
-        return other._names == self._names and other.subgroups == self.subgroups
+        return other.all == self.all or (
```

Comment:
if `other.all` and `self.all` are both `False`, this will return `True`, ignoring the logic in the parentheses. 

I think this might want to be 
```python
    return (other.all and self.all) or (
        other._names == self._names and other.subgroups == self.subgroups
    )
```